core/domain/classroom_config_services.py

```diff
@@ -175,23 +188,43 @@ def get_new_classroom_id() -> str:
 
 
 def update_classroom(
-    classroom: classroom_config_domain.Classroom,
-    classroom_model: classroom_models.ClassroomModel
+    classroom: classroom_config_domain.Classroom
 ) -> None:
     """Saves a Clasroom domain object to the datastore.
 
     Args:
         classroom: Classroom. The classroom domain object for the given
             classroom.
-        classroom_model: ClassroomModel. The classroom model instance.
     """
     classroom.validate()
+    classroom_model = classroom_models.ClassroomModel.get(
+        classroom.classroom_id, strict=False)
+
+    if not classroom_model:
+        return
+
     classroom_model.name = classroom.name
     classroom_model.url_fragment = classroom.url_fragment
     classroom_model.course_details = classroom.course_details
     classroom_model.topic_list_intro = classroom.topic_list_intro
     classroom_model.topic_id_to_prerequisite_topic_ids = (
         classroom.topic_id_to_prerequisite_topic_ids)
+    classroom_model.teaser_text = classroom.teaser_text
+    classroom_model.is_published = classroom.is_published
+    classroom_model.thumbnail_filename = (
+        classroom.thumbnail_data['filename']
```

Comment:
Shouldn't you be using a plain Python domain object for this, so that access is by classroom.thumbnail_data.filename? See e.g. the sub-objects for the Exploration class.

Ditto elsewhere. In general try not to use dicts except at the boundaries of the system (e.g. connection to datastore, or connection to frontend).

Reply:
My bad. Updated the code.